src/net/http.rs

```diff
@@ -1143,28 +1177,120 @@ impl HttpRequestType {
+        if preamble.get_content_length() != 0 {
+            return Err(net_error::DeserializeError("Invalid Http request: expected 0-length body for GetTransferCost".to_string()));
+        }
+
+        Ok(HttpRequestType::GetTransferCost(HttpRequestMetadata::from_preamble(preamble)))
+    }
+
+    fn parse_get_account<R: Read>(_protocol: &mut StacksHttp, preamble: &HttpRequestPreamble, captures: &Captures, _fd: &mut R) -> Result<HttpRequestType, net_error> {
+        if preamble.get_content_length() != 0 {
+            return Err(net_error::DeserializeError("Invalid Http request: expected 0-length body for GetAccount".to_string()));
+        }
+
+        let principal = PrincipalData::parse(&captures["principal"])
+            .map_err(|_e| net_error::DeserializeError("Failed to parse account principal".into()))?;
+
+        Ok(HttpRequestType::GetAccount(HttpRequestMetadata::from_preamble(preamble), principal))
+    }
+
+    fn parse_get_map_entry<R: Read>(_protocol: &mut StacksHttp, preamble: &HttpRequestPreamble, captures: &Captures, fd: &mut R) -> Result<HttpRequestType, net_error> {
+        let content_len = preamble.get_content_length();
+        if !(content_len > 0 && content_len < (MAX_VALUE_SIZE * 4)) {
+            return Err(net_error::DeserializeError("Invalid Http request: invalid body length for GetMapEntry".to_string()));
+        }
+
+        let contract_addr =  StacksAddress::from_string(&captures["address"])
+            .ok_or_else(|| net_error::DeserializeError("Failed to parse contract address".into()))?;
+        let contract_name = ContractName::try_from(captures["contract"].to_string())
+            .map_err(|_e| net_error::DeserializeError("Failed to parse contract name".into()))?;
+        let map_name = ClarityName::try_from(captures["map"].to_string())
+            .map_err(|_e| net_error::DeserializeError("Failed to parse contract name".into()))?;
+
+        let value_hex: String = serde_json::from_reader(fd)
+            .map_err(|_e| net_error::DeserializeError("Failed to parse JSON body".into()))?;
+
+        let value = Value::try_deserialize_hex_untyped(&value_hex)
+            .map_err(|_e| net_error::DeserializeError("Failed to deserialize key value".into()))?;
+
+        Ok(HttpRequestType::GetMapEntry(HttpRequestMetadata::from_preamble(preamble), contract_addr, contract_name, map_name, value))
+    }
+
+    fn parse_call_read_only<R: Read>(_protocol: &mut StacksHttp, preamble: &HttpRequestPreamble, captures: &Captures, fd: &mut R) -> Result<HttpRequestType, net_error> {
+        let content_len = preamble.get_content_length();
+        if !(content_len > 0 && content_len < (MAX_VALUE_SIZE * 4 * 20)) {
+            return Err(net_error::DeserializeError("Invalid Http request: invalid body length for GetMapEntry".to_string()));
+        }
+
```

Comment:
Shouldn't this have content-type `application/json`?

Reply:
Okay, added a check in 748a98b40ee2c0bd53272e5e468bf180074573b8